proof-systems/src/darlin/t_dlog_acc_marlin/iop/verifier.rs

```diff
@@ -0,0 +1,199 @@
+            ))?
+        }
+
+        let msg = VerifierSecondMsg { beta };
+        state.second_round_msg = Some(msg);
+
+        Ok((msg, state))
+    }
+
+    /// Third round of the verifier, samples the random challenges `gamma` and `lambda` for the
+    /// inner sumcheck aggregation.
+    pub fn verifier_third_round<'a, R: FiatShamirRng>(
+        mut state: VerifierState<'a, G1, G2>,
+        fs_rng: &mut R,
+    ) -> Result<(VerifierThirdMsg<G1::ScalarField>, VerifierState<'a, G1, G2>), Error> {
+        let gamma: G1::ScalarField = fs_rng.squeeze_128_bits_challenge();
+        if state
+            .domain_h
+            .evaluate_vanishing_polynomial(gamma)
+            .is_zero()
+        {
+            Err(Error::Other(
+                "Sampled a gamma challenge belonging to H domain".to_owned(),
+            ))?
+        }
+
+        let lambda: G1::ScalarField = fs_rng.squeeze_128_bits_challenge();
+
+        let msg = VerifierThirdMsg { gamma, lambda };
+        state.third_round_msg = Some(msg);
+
+        Ok((msg, state))
+    }
+
+    /// Output the query state and next round state.
+    pub fn verifier_query_set<'a, 'b>(
+        state: VerifierState<G1, G2>,
+    ) -> Result<(QuerySet<'b, G1::ScalarField>, VerifierState<G1, G2>), Error> {
+        if state.second_round_msg.is_none() {
+            return Err(Error::Other("Second round message is empty".to_owned()));
+        }
+        let beta = state.second_round_msg.unwrap().beta;
+        let alpha = state.first_round_msg.as_ref().unwrap().alpha;
+        let prev_alpha = state.previous_inner_sumcheck_acc.1.alpha;
+        let gamma = state.third_round_msg.unwrap().gamma;
+
+        let g_h = state.domain_h.group_gen();
+
+        let mut query_set = QuerySet::new();
+
+        // First round polys
+        query_set.insert(("x".into(), ("beta".into(), beta)));
+        query_set.insert(("w".into(), ("beta".into(), beta)));
+        query_set.insert(("y_a".into(), ("beta".into(), beta)));
+        query_set.insert(("y_b".into(), ("beta".into(), beta)));
+
+        // Second round polys
+        query_set.insert(("u_1".into(), ("beta".into(), beta)));
+        query_set.insert(("u_1".into(), ("g * beta".into(), g_h * beta)));
+        query_set.insert(("h_1".into(), ("beta".into(), beta)));
+        query_set.insert(("t".into(), ("beta".into(), beta)));
+
+        // Inner sumcheck aggregation polys
+        query_set.insert(("curr_bridging_poly".into(), ("alpha".into(), alpha)));
+        query_set.insert((
+            "prev_bridging_poly".into(),
+            ("prev_alpha".into(), prev_alpha),
+        ));
+        query_set.insert(("curr_bridging_poly".into(), ("gamma".into(), gamma)));
+        query_set.insert(("prev_bridging_poly".into(), ("gamma".into(), gamma)));
+        query_set.insert(("curr_t_acc_poly".into(), ("beta".into(), beta)));
+        query_set.insert(("prev_t_acc_poly".into(), ("beta".into(), beta)));
```

Comment:
After the changes in poly-commit `QueryMap`, it is easier and clearer to build the verifier `query_map` by aggregating queries by query point rather than by protocol round, as previously done. This would make the suggested comments a bit awkward to insert, in my opinion. Rather than trying a half-hearted attempt to explain the logic of the checks inside comments, we could instead provide a reference to the appropriate section of the Darlin paper. What do you think?